lesson6-questions.txt

```diff
@@ -3,45 +3,83 @@ Questions for Lesson 6
+ URL pattern: The URL path for the route.
+Controller and Action: This indicates which controller and action should handle the request.
 
 
 3. What does the Rails db:migrate command do?
-
-
+  This command will apply all pending migrations in the db/migrate directory to the database, bringing it up to date with the latest schema changes
 
 4. What file do you look at to see the current database schema?
-
+To see the current database schema in a Rails application, you look at the db/schema.rb file.
 
 5. What command shows the routes?
 
+bin/rails routes
 
 6. How can you start the Rails console? How can you do CRUD operations from the console?
-
+bin/rails c
+create -> user = User.create(name: '', email: '')
+read all -> users = User.all
+read one by id -> user = User.find(1)
+update -> user = User.find(1)  +    user.update(name: 'MAry Li')
+delete -> user = User.find(1)  +    user.destroy
 
 7. What statements are in erb files that make them different from html files?  What are the
 two kinds of special statements, and how are they marked in the erb file? Where are they executed?
 
+The Ruby code within these erb statements is executed on the server side when the template is rendered. 
+The result of this execution is plain HTML, which is then sent to the client's browser.
+there are two kinds of special statements in ERB files:
+Output Statements (<%= ... %>): Insert the result of Ruby code into the HTML.
+Script Statements (<% ... %>): Execute Ruby code without inserting anything into the HTML.
 
 8. Consider this statement: @forum = Forum.new().  Why is Forum capitalized?  Why is @forum not
 capitalized? Why is the @ used?
 
-
+Forum is capitalized because it is a class name in Ruby. 
+@forum is an instance variable, and instance variables in Ruby begin with an @ symbol and are written in lowercase (snake_case). 
+The @ symbol indicates that forum is an instance variable.
 
 9. Consider these statements:
 
 @forum = Forum.new(...)
 @forum =Forum.create(...)
 @forum.save(...)
 
-What is the difference between Forum.new() and Forum.create()?  Why can't you do Forum.save(...)?
+@forum = Forum.new(...) ->
+ This statement creates a new instance of the Forum class with the attributes specified in the parentheses.
+
+@forum =Forum.create(...) -> 
+This statement creates a new instance of the Forum class and immediately saves it to the database. It combines new and save in one step.
 
+@forum.save(...) ->
+ this statement saves the current instance of @forum to the database. This method is called on an object that has already been instantiated, 
+ either by new or by loading it from the database.
+
+What is the difference between Forum.new() and Forum.create()?  Why can't you do Forum.save(...)?
+Forum.new(...): Instantiates a new Forum object but does not save it to the database.
+Forum.create(...): Instantiates and saves a new Forum object to the database in one step.
```

Comment:
This is great! But I don't see an answer to the question `Why can't you do Forum.save(...)?` (Note that `Forum.new`, `Forum.create` are permissible, but not `Forum.save`...only `@forum.save`). We can talk through this in our 1:1 if you'd like!